src/porepy/numerics/fv/mpfa.py

```diff
@@ -870,7 +904,116 @@ def _local_discr(
+                etc.
+            bound_exclusion: Object that can eliminate faces related to boundary
+                conditions.
+            darcy: discretized Darcy's law
+            igrad: inverse gradient
+            nk_grad_all: nK products on a sub-cell level
+            nk_grad_paired: nK products after pairing
+            
+        Returns:
+            scipy.sparse.csr_matrix (shape num_subfaces, num_subcells):
+            discretization of the gravity term, interpreted as a force on a subface
+            due to imbalance in cell-center gravities
+
+        Method properties and implementation details.
+        Basis functions, namely 'flux' and 'bound_flux', for the pressure
+        discretization are obtained as in standard MPFA-O method.
+        Gravity is represented as forces in the cells.
+        However, jumps in gravitational forces over a cell face act as flux
+        imbalance, and thus induce additional pressure gradients in the sub-cells.
+        An additional system is set up, which applies non-zero conditions to the
+        flux continuity equation. This can be expressed as a linear system on the form
+            (i)   A * grad_p            = I
+            (ii)  B * grad_p + C * p_cc = 0
+            (iii) 0            D * p_cc = 0
+        Thus (i)-(iii) can be inverted to express the additional pressure gradients
+        due to imbalance in gravitational forces as in terms of the cell center variables.
+        Thus we can compute the basis functions 'div_g_jumps' on the sub-cells.
+        To ensure flux continuity, as soon as a convention is chosen for what side
+        the flux evaluation should be considered on, an additional term, called
+        'div_g_face', is added to the full flux. This latter term represents the flux
+        due to cell-center gravity acting on the face from the chosen side.
+        The pair subfno_unique-unique_subfno gives the side convention.
+        The full flux on the face is therefore given by
+        q = flux * p + bound_flux * p_b + (div_g_jumps + div_g_face) * g
+
+        The strategy is as follows.
+        1. assemble r.h.s. for the new linear system, needed for the term 'div_g_jumps'
+        2. compute term 'div_g_face'
+        """
+
+        num_subfno = subcell_topology.num_subfno
+        num_subfno_unique = subcell_topology.subfno_unique.size
+
+        # Step 1
+        # The gravity term in the flux continuity equation is discretized
+        # as a force on the faces. The right hand side is thus formed of the
+        # unit vector.           
+        vals = np.ones(num_subfno_unique)
+        rows = subcell_topology.subfno_unique 
+        cols = subcell_topology.subfno_unique
+        rhs_units = sps.coo_matrix((vals, (rows, cols)), 
+                                 shape=(num_subfno_unique,
+                                        num_subfno_unique))
+
+        rhs_units = bound_exclusion.exclude_dirichlet(rhs_units)
```

Comment:
Good catch. I belive this will not work; the ordering of the equations change when applying a Robin condition (see line 797 in mpfa and comment above). I made a commit that I think fixes this

Reply:
perfect I integrated it and works! Thanks